ggml.h

```diff
@@ -227,7 +227,7 @@
 #define GGML_MAX_DIMS           4
 #define GGML_MAX_PARAMS         2048
 #define GGML_MAX_CONTEXTS       64
-#define GGML_MAX_SRC            10
+#define GGML_MAX_SRC            62
```

Comment:
Hm, we are not well prepared for 60-expert MoE models

This change will more than double the size of `ggml_tensor` and I still don't know how to support many source buffers in Metal:

https://github.com/ggerganov/llama.cpp/blob/131b0584096ee9df4d07cb28759dfea6efe6475f/ggml-metal.m#L1750-L1759

What are our options here? cc @slaren

Reply:
The size of each expert is very small. The ffn_moe_intermediate size is divided by 4.

Reply:
We could store all the experts in the same tensor, I think it would be easy to adapt the implementations.

Reply:
Should we merge this PR first and then try to resolve this?

Reply:
I am not sure if we need to merge this right now. As it is, this will only work with the CPU backend, the sort operators in the GPU backends also require the number of experts to be a power of two, so assuming that it is a large model, this implementation may not be very useful. When we have a model to test with, we can improve the implementation.

Reply:
I remember qwen2 has 1.8BX16. that make sense

Reply:
@slaren do you have some implementations to deal with this **extra large number** experts issue?